src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/Public.Data/DataSetsControllerTests.cs

```diff
@@ -493,4 +493,382 @@ private static ClaimsPrincipal UserWithAccessAllPublicationsClaim()
+
+            DataSetVersion liveDataSetVersion = DataFixture
+                .DefaultDataSetVersion(
+                    filters: 1,
+                    indicators: 1,
+                    locations: 1,
+                    timePeriods: 2)
+                .WithStatusPublished()
+                .WithCsvFileId(liveFile.Id)
+                .WithDataSet(dataSet)
+                .FinishWith(dsv => dsv.DataSet.LatestLiveVersion = dsv);
+
+            DataSetVersion draftDataSetVersion = DataFixture
+                .DefaultDataSetVersion(
+                    filters: 1,
+                    indicators: 1,
+                    locations: 1,
+                    timePeriods: 2)
+                .WithStatusDraft()
+                .WithCsvFileId(draftFile.Id)
+                .WithDataSet(dataSet)
+                .FinishWith(dsv => dsv.DataSet.LatestDraftVersion = dsv);
+
+            await TestApp.AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersions.AddRange(liveDataSetVersion, draftDataSetVersion);
+                context.DataSets.Update(dataSet);
+            });
+
+            var response = await GetDataSet(dataSet.Id);
+
+            var content = response.AssertOk<DataSetSummaryViewModel>();
+
+            Assert.NotNull(content);
+            Assert.Equal(dataSet.Id, content.Id);
+            Assert.Equal(dataSet.Title, content.Title);
+            Assert.Equal(dataSet.Summary, content.Summary);
+            Assert.Equal(dataSet.Status, content.Status);
+            Assert.Equal(liveDataSetVersion.Id, content.LatestLiveVersion!.Id);
+            Assert.Equal(liveDataSetVersion.Version, content.LatestLiveVersion.Version);
+            Assert.Equal(liveDataSetVersion.Status, content.LatestLiveVersion.Status);
+            Assert.Equal(liveDataSetVersion.VersionType, content.LatestLiveVersion.Type);
+            Assert.Equal(liveFile.DataSetFileId, content.LatestLiveVersion.DataSetFileId);
+            Assert.Equal(liveReleaseVersion.Id, content.LatestLiveVersion.ReleaseVersion.Id);
+            Assert.Equal(liveReleaseVersion.Title, content.LatestLiveVersion.ReleaseVersion.Title);
+            Assert.Equal(draftDataSetVersion.Id, content.DraftVersion!.Id);
+            Assert.Equal(draftDataSetVersion.Version, content.DraftVersion.Version);
+            Assert.Equal(draftDataSetVersion.Status, content.DraftVersion.Status);
+            Assert.Equal(draftDataSetVersion.VersionType, content.DraftVersion.Type);
+            Assert.Equal(draftFile.DataSetFileId, content.DraftVersion.DataSetFileId);
+            Assert.Equal(draftReleaseVersion.Id, content.DraftVersion.ReleaseVersion.Id);
+            Assert.Equal(draftReleaseVersion.Title, content.DraftVersion.ReleaseVersion.Title);
+        }
+
+        [Fact]
+        public async Task RequestedDataSetHasNoDraftVersion_Returns200_NoDraftVersion()
```

Comment:
Here the ReleaseFile is set up for the published ReleaseVersion, but not the draft ReleaseVersion.
That won't usually be the case, because when a new release version is created, the files associated with its predecessor a linked to it. So at least to begin with. the draft release version has all the files of the published release version.

I think this test needs adjusting for this.

I think it might expose a problem just prior to building the view model where the ReleaseFiles associated with the published and draft data set versions are collected into a `Dictionary<Guid,ReleaseFile> keyed by file id. The published dataset will be for a file id that's associated with two different ReleaseFiles.

Reply:
Discussed

There was a bug due to looking for `ReleaseFiles` associated with the same `FileId`

This has now been fixed, by using the new `DataSetVersion.ReleaseFileId` association